File: rockfish/tests/issues/negativeHeaderCoords/read_header.py
import obspy.segy.core as core
import obspy.segy.segy as segy

filename = '/Users/ncm/Rockfish/rockfish/tests/data/pos_neg.segy'

hdrwords = ['source_coordinate_x', 'source_coordinate_y',
           'group_coordinate_x', 'group_coordinate_y']

print "With obspy.segy.segy.readSEGY:"
sgy = segy.readSEGY(filename,unpack_headers=True)
for k in hdrwords:
    print k,":",sgy.traces[0].header.__getattribute__(k)

print "\nWith obspy.segy.core.readSEGY:"
st = core.readSEGY(filename,unpack_trace_headers=True)
tr = st[0]
for k in hdrwords:
    print k,":",st[0].stats.segy.trace_header[k]

File: rockfish/tests/issues/negativeHeaderCoords/read_header.m


addpath('~/sw/SegyMAT')


isegy = '/Users/ncm/Rockfish/rockfish/tests/data/pos_neg.segy';

[D,TH,SH]=ReadSegy(isegy);

i=1;
fprintf('TH(%i).SourceX: %f\n',i,TH(i).SourceX)
fprintf('TH(%i).SourceY: %f\n',i,TH(i).SourceY)
fprintf('TH(%i).GroupX: %f\n',i,TH(i).GroupX)
fprintf('TH(%i).GroupY: %f\n',i,TH(i).GroupY)
